File: Client/Android/solutions/interactivelive/src/main/java/com/vertcdemo/solution/interactivelive/util/CenteredToast.java
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.vertcdemo.solution.interactivelive.util;

import android.app.Application;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.MainThread;
import androidx.annotation.StringRes;

import com.vertcdemo.solution.interactivelive.R;
import com.vertcdemo.core.utils.AppUtil;

public class CenteredToast {
    @MainThread
    public static void show(String message) {
        final Application context = AppUtil.getApplicationContext();
        final TextView view = (TextView) LayoutInflater.from(context).inflate(R.layout.toast_centered, null);
        view.setText(message);
        final Toast toast = new Toast(context);
        toast.setGravity(Gravity.CENTER, 0, 0);
        toast.setView(view);
        toast.show();
    }

    public static void show(@StringRes int message) {
        show(message, Toast.LENGTH_SHORT);
    }

    public static void show(@StringRes int message, int duration) {
        final Application context = AppUtil.getApplicationContext();
        final TextView view = (TextView) LayoutInflater.from(context).inflate(R.layout.toast_centered, null);
        view.setText(message);
        final Toast toast = new Toast(context);
        toast.setGravity(Gravity.CENTER, 0, 0);
        toast.setView(view);
        toast.setDuration(duration);
        toast.show();
    }
}

File: Client/Android/component/solution-base/src/main/java/com/vertcdemo/ui/CenteredToast.kt
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0
package com.vertcdemo.ui

import android.view.Gravity
import android.view.LayoutInflater
import android.widget.TextView
import android.widget.Toast
import androidx.annotation.MainThread
import androidx.annotation.StringRes
import com.vertcdemo.base.R
import com.vertcdemo.core.utils.AppUtil.applicationContext

@Suppress("DEPRECATION")
object CenteredToast {
    @JvmStatic
    @JvmOverloads
    @MainThread
    fun show(message: String, duration: Int = Toast.LENGTH_SHORT) {
        Toast(applicationContext).apply {
            setGravity(Gravity.CENTER, 0, 0)
            val view =
                LayoutInflater.from(applicationContext)
                    .inflate(R.layout.toast_centered, null) as TextView
            view.text = message
            this.view = view
            this.duration = duration
        }.also {
            it.show()
        }
    }

    @JvmStatic
    @JvmOverloads
    @MainThread
    fun show(@StringRes message: Int, duration: Int = Toast.LENGTH_SHORT) {
        Toast(applicationContext).apply {
            setGravity(Gravity.CENTER, 0, 0)
            val toast =
                LayoutInflater.from(applicationContext)
                    .inflate(R.layout.toast_centered, null) as TextView
            toast.setText(message)
            view = toast
            this.duration = duration
        }.also {
            it.show()
        }
    }
}
